contracts/StorageAccessible.sol

```diff
@@ -23,4 +23,41 @@ contract StorageAccessible {
         }
         return result;
     }
+
+    /**
+     * @dev Performs a delegetecall on a targetContract in the context of self.
+     * Internally reverts execution to avoid side effects (making it static). Catches revert and returns encoded result as bytes.
+     * @param targetContract Address of the contract containing the code to execute.
+     * @param calldataPayload Calldata that should be sent to the target contract (encoded method name and arguments).
+     */
+    function executeStaticDelegatecall(
+        address targetContract,
+        bytes memory calldataPayload
+    ) public returns (bytes memory) {
+        bytes memory innerCall = abi.encodeWithSignature(
+            "executeStaticDelegatecallInternal(address,bytes)",
+            targetContract,
+            calldataPayload
+        );
+        (, bytes memory response) = address(this).call(innerCall);
+        return response;
+    }
+
+    /**
+     * @dev Performs a delegetecall on a targetContract in the context of self.
+     * Internally reverts execution to avoid side effects (making it static). Returns encoded result as revert message.
+     * @param targetContract Address of the contract containing the code to execute.
+     * @param calldataPayload Calldata that should be sent to the target contract (encoded method name and arguments).
+     */
+    function executeStaticDelegatecallInternal(
+        address targetContract,
+        bytes memory calldataPayload
+    ) public returns (bytes memory) {
+        (, bytes memory response) = targetContract.delegatecall(
+            calldataPayload
+        );
+        assembly {
+            revert(add(response, 0x20), mload(response))
```

Comment:
Should we try to open a issue with the solidity team to get a way to do this without assembly (the revert of bytes)?

Reply:
Clever! Do I understand this to mean that the response data is the revert message?

Reply:
So a revert message works different. Solidity encoded the revert message like a call to a function `Error(string)` (see https://solidity.readthedocs.io/en/v0.5.3/control-structures.html#error-handling-assert-require-revert-and-exceptions). Currently it is not possible to returns raw bytes (without any additional encoding). Therefore it is required to use assembly for this.

@fleupold if required I could open an issue on the solidity repo

Reply:
So the issue would request support for calling `revert` with bytes as arguments inside solidity right? I'm not sure this is the intended use of revert (we are kind of shoehorning it a bit here) and thus I'm fine with using assembly for this.

Do you have another use case where we would legitimately want to return bytes from a revert?

Feel free to open issue regardless, we can see what the solidity community thinks about.